cmd/gardenlet/app/migration.go

```diff
@@ -6,10 +6,150 @@ package app
 
 import (
 	"context"
+	"fmt"
+	"slices"
 
 	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	gardeneraccess "github.com/gardener/gardener/pkg/component/gardener/access"
+	shootsystem "github.com/gardener/gardener/pkg/component/shoot/system"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
 )
 
-func (g *garden) runMigrations(_ context.Context, _ logr.Logger) error {
+func (g *garden) runMigrations(ctx context.Context, log logr.Logger) error {
+	log.Info("Migrating ClusterRoleBindings for shoot/adminkubeconfig and shoot/viewerkubeconfig")
+	if err := migrateAdminViewerKubeconfigClusterRoleBindings(ctx, log, g.mgr.GetClient()); err != nil {
+		return fmt.Errorf("failed migrating ClusterRoleBindings for shoot/adminkubeconfig and shoot/viewerkubeconfig: %w", err)
+	}
+
 	return nil
 }
+
+// migrateAdminViewerKubeconfigClusterRoleBindings moves the ClusterRoleBindings granting access to the
+// shoot/adminkubeconfig and shoot/viewerkubeconfig subresources from the shoot-core-system managed resource to the
+// shoot-core-gardeneraccess managed resource.
+// TODO(vpnachev): Remove this after v1.132.0 has been released.
```

Comment:
```suggestion
// TODO(vpnachev): Remove this after v1.133.0 has been released.
```

We should keep such internal migration code for at least 3 releases: https://github.com/gardener/gardener/blob/6c9be31f366443244831665c9a9dd607f814fa00/docs/development/process.md#L236-L238
I am not sure if it will be fair to count the patch releases and expect everyone upgrades first to them before upgrading to the next minor version.
Having in mind that 1.131.0 will be the first minor release to include this fix, then we should remove after 1.133.0 has been released.